Jellyfin.Data/Entities/Image.cs

```diff
@@ -0,0 +1,106 @@
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using Jellyfin.Data.Enums;
+using Jellyfin.Data.Interfaces;
+
+namespace Jellyfin.Data.Entities.Libraries
+{
+    /// <summary>
+    /// An entity representing a single image.
+    /// </summary>
+    public class Image : IHasConcurrencyToken
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Image"/> class.
+        /// </summary>
+        /// <param name="path">The path of the image.</param>
+        /// <param name="type">The image type.</param>
+        public Image(string path, ImageType type = ImageType.Primary)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            Path = path;
+            Type = type;
+            AddedDate = DateTime.UtcNow;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Image"/> class.
+        /// </summary>
+        /// <param name="path">The path of the image.</param>
+        /// <param name="type">The image type.</param>
+        /// <param name="lastModifiedDate">The last modification date.</param>
+        public Image(string path, DateTime lastModifiedDate, ImageType type = ImageType.Primary) : this(path, type)
+        {
+            LastModifiedDate = lastModifiedDate;
+        }
+
+        /// <summary>
+        /// Gets the id.
+        /// </summary>
+        /// <remarks>
+        /// Identity, Indexed, Required.
+        /// </remarks>
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int Id { get; private set; }
+
+        /// <summary>
+        /// Gets the path.
+        /// </summary>
+        /// <remarks>
+        /// Required.
+        /// </remarks>
+        public string Path { get; private set; }
+
+        /// <summary>
+        /// Gets the image type.
+        /// </summary>
+        /// <remarks>
+        /// Required.
+        /// </remarks>
+        public ImageType Type { get; private set; }
+
+        /// <summary>
+        /// Gets the blurhash string.
+        /// </summary>
+        public string? Blurhash { get; private set; }
+
+        /// <summary>
+        /// Gets the date last modified.
+        /// </summary>
+        /// <remarks>
+        /// Required.
+        /// </remarks>
+        public DateTime LastModifiedDate { get; private set; } = DateTime.UtcNow;
+
+        /// <summary>
+        /// Gets the date when the image was added to the database.
+        /// </summary>
+        public DateTime AddedDate { get; private set; }
+
+        /// <summary>
+        /// Gets the creation date of the file.
+        /// </summary>
+        public DateTime? FileCreationDate { get; private set; }
+
+        /// <summary>
+        /// Gets the modification date of the file.
+        /// </summary>
+        public DateTime? FileModificationDate { get; private set; }
```

Comment:
I just meant these items are named somewhat inconsistently. One option would be the one I suggested yesterday, and another one might be combining the two modified properties (they seem a bit redundant) and removing the `File` prefix completely.

Reply:
Right now ofc they're redundant but I can see when this can be useful once everything uses this entity. Examples:

* **Change the type of an image**: This is currently impossible with the API, but we can have this in the future, and show the date of that change in the UI. The change of the type of an image doesn't imply that Blurhash changes, so the computation can be skipped

* **File modified on disk**: We could have modified the properties of the image on the UI, but also some people might edit the library images in place (or use a compression script to reduce their sizes), which means that we need to recompute their blurhashes, which is something we can do if we detect a mismatch during scanning time and the database value. Also, if the scanner gets interrupted when this situation happen, the scanner will be able to catch up as it will detect that the database last modified date is older than the file's.

The added dates I can agree with you that are less useful. I just put them there so we can expose that information in the API the same way we do with media items.